### ISSUE TITLE
Zero-Copy Non-Owning Read-Only String Views in libosrm API

### ISSUE THREAD

**Issue by daniel-j-h** [Project-OSRM/osrm-backend#3591]
With https://github.com/Project-OSRM/osrm-backend/pull/3459 the facades return string views now. String views are non-owning read-only zero-copy string accessors to the underlying block of memory.

This is great, since we no longer copy strings around having to pay for unnecessary allocations when the strings are fixed in memory anyway.

To fully benefit from it not only _inside_ the engine but also e.g. on the node-osrm / libosrm boundary, we should expose string views in the libosrm API, too.

Then users can e.g. create Javascript objects directly from libosrm's large chunk of memory (either internal or shared mem) instead of doing the `mem -> std::string -> v8::String` dance.

Note: string views are non-owning which makes handling lifetimes hard(er) for users. Think: responses outliving the osrm object. This won't be possible since the osrm object will then own the memory for the response strings. cc @TheMarex

**Comment by TheMarex** [comment]
> Note: string views are non-owning which makes handling lifetimes hard(er) for users. Think: responses outliving the osrm object. This won't be possible since the osrm object will then own the memory for the response strings. 

Good catch. Yes this is basically a deal-breaker right now since we only guarantee that the `DataFacade` lives as long as the query. If the result is returned from OSRM this could lead to invalid references.

**Comment by DennisOSRM** [comment]
Closing since this is not memory safe.
